accelerateStrategy.cpp

```diff
@@ -0,0 +1,229 @@
+/* Copyright 2017 Konstantin Batoev.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This file was modified by Konstantin Batoev to make it comply with the requirements of trikRuntime
+ * project. See git revision history for detailed changes. */
+
+#include "accelerateStrategy.h"
+
+#include <math.h>
+
+AccelerateStrategy::AccelerateStrategy()
+	: pad1WasActive(false)
+	, pad2WasActive(false)
+	, speed(300)
+	, padsMapper(new QSignalMapper(this))
+{
+
+	connect(&stopTimerForPad1, SIGNAL(timeout()), padsMapper, SLOT(map()));
+	padsMapper->setMapping(&stopTimerForPad1, 1);
+	connect(&stopTimerForPad1, SIGNAL(timeout()), &stopTimerForPad1, SLOT(stop()));
+
+	connect(&stopTimerForPad2, SIGNAL(timeout()), padsMapper, SLOT(map()));
```

Comment:
There was one pad for both pads in the previous version. The problem was when I pressed pad1, then pressed pad2, released pad1, and released pad2 in some time (ex. 30 seconds). And only in the end there were two commands: pad 1 up, pad 2 up, but I expected command "pad 1 up", when I released pad1.
So, I decided to use timer per pad.